pkg/liquidity-mining/contracts/VotingEscrowRemapper.sol

```diff
@@ -0,0 +1,306 @@
+        _localToRemoteAddressMap[chainId][localUser] = remoteUser;
+
+        // Note: it is important to perform the bridge calls _after_ the mappings are settled, since the
+        // omni voting escrow will rely on the correct mappings to bridge the balances.
+        (uint256 nativeFee, ) = omniVotingEscrow.estimateSendUserBalance(uint16(chainId), false, "");
+        if (oldRemoteUser != address(0)) {
+            // If there was an old mapping, send balance from (local) oldRemoteUser --> (remote) oldRemoteUser
+            // This should clean up the existing bridged balance from localUser --> oldRemoteUser.
+            omniVotingEscrow.sendUserBalance{ value: nativeFee }(
+                oldRemoteUser,
+                uint16(chainId),
+                payable(msg.sender),
+                address(0),
+                ""
+            );
+        }
+
+        // Bridge balance for new mapping localUser --> remoteUser.
+        omniVotingEscrow.sendUserBalance{ value: nativeFee }(
+            localUser,
+            uint16(chainId),
+            payable(msg.sender),
+            address(0),
+            ""
+        );
+
+        emit AddressMappingUpdated(localUser, remoteUser, uint16(chainId));
+        // Send back any leftover ETH to the caller.
+        uint256 remainingBalance = address(this).balance;
+        if (remainingBalance > 0) {
+            Address.sendValue(msg.sender, remainingBalance);
+        }
+    }
+
+    /**
+     * @notice Sets an address to manage the mapping for a given local user on its behalf.
+     * @dev This is intended to handle contracts which cannot interact with this contract directly.
+     * @param localUser - The address of a contract allowlisted on the `SmartWalletChecker`.
+     * @param delegate - The address which is allowed to manage remote users to be linked to `localUser`.
+     */
+    function setNetworkRemappingManager(address localUser, address delegate) external authenticate {
+        require(_isAllowedContract(localUser), "Only contracts which can hold veBAL may have a delegate");
+
+        _localRemappingManager[localUser] = delegate;
+        emit AddressDelegateUpdated(localUser, delegate);
+    }
+
+    /**
+     * @notice Clears a local user's mapping for a particular network.
+     * @dev This is intended to discourage and also allow recovery from griefing attacks.
+     * If griefing occurs then the griefer can be removed from Smart Wallet Checker and have their remappings erased.
+     * The local user can always clear its own mapping, regardless the state of the Smart Wallet Checker.
+     * @param localUser - The address of the local user to erase.
+     * @param chainId - The chain id of the network to erase.
+     */
+    function clearNetworkRemapping(address localUser, uint256 chainId) external payable nonReentrant {
+        require(!_isAllowedContract(localUser) || localUser == msg.sender, "localUser is still in good standing");
```

Comment:
How about this? https://github.com/balancer/balancer-v2-monorepo/pull/2174/commits/ec210abac0278da1d5195cfbb8dc895eb9d930fc

Leaving the manager should be harmless, since nor the user nor the manager can set a mapping again if the address is not allow listed in the smart wallet checker.

The manager cleaning 'feature' was not even documented, and it doesn't make much sense in this context as you say.